lib/Platform/Intl/java/com/facebook/hermes/intl/PlatformNumberFormatterICU.java

```diff
@@ -0,0 +1,358 @@
+        }
+        if (attribute == DecimalFormat.Field.EXPONENT_SYMBOL) {
+            return "exponentSeparator";
+        }
+        if (attribute == DecimalFormat.Field.DECIMAL_SEPARATOR) {
+            return "decimal";
+        }
+        if (attribute == DecimalFormat.Field.GROUPING_SEPARATOR) {
+            return "group";
+        }
+        if (attribute == DecimalFormat.Field.PERCENT) {
+            return "percentSign";
+        }
+        if (attribute == DecimalFormat.Field.PERMILLE) {
+            return "permilleSign";
+        }
+        if (attribute == DecimalFormat.Field.CURRENCY) {
+            return "currency";
+        }
+        // Report unsupported/unexpected number fields as literal.
+        return "literal";
+    }
+
+    @Override
+    public AttributedCharacterIterator formatToParts(double n) throws JSRangeErrorException {
+        AttributedCharacterIterator iterator;
+        if (mFinalFormat instanceof MeasureFormat && mUnitId != null) {
+
+            if (mMeasureUnit == null)
+                mMeasureUnit = parseUnit(mUnitId);
+
+            iterator = mFinalFormat.formatToCharacterIterator(new Measure(n, mMeasureUnit));
+        } else {
+            iterator = mFinalFormat.formatToCharacterIterator(n);
+        }
+
+        return iterator;
+    }
+
+    public static int getCurrencyDigits(String currencyCode) throws JSRangeErrorException {
+        try {
+            return Currency.getInstance(currencyCode).getDefaultFractionDigits();
+        } catch (IllegalArgumentException ex) {
+            throw new JSRangeErrorException("Invalid currency code !");
+        }
+    }
+
+    public static PlatformNumberFormatterICU createDecimalFormat(ILocaleObject localeObject, IPlatformNumberFormatter.Style style,
+                                                                 IPlatformNumberFormatter.CurrencySign currencySign,
+                                                                 IPlatformNumberFormatter.Notation notation,
+                                                                 IPlatformNumberFormatter.CompactDisplay compactDisplay) throws JSRangeErrorException {
+        PlatformNumberFormatterICU platformNumberFormatterICU;
+        switch (style) {
+            case CURRENCY:
+                platformNumberFormatterICU = PlatformNumberFormatterICU.createDecimalFormatForCurrency(localeObject, currencySign, style);
+                break;
+            case PERCENT:
+                platformNumberFormatterICU = PlatformNumberFormatterICU.createDecimalFormatForPercent(localeObject, style);
+                break;
+            case DECIMAL:
+            case UNIT:
+                platformNumberFormatterICU = PlatformNumberFormatterICU.createDecimalFormatGerneric(localeObject, notation, compactDisplay, style);
+                break;
+            default:
+                throw new JSRangeErrorException("Unrecognized formatting style requested.");
+        }
```

Comment:
This pattern using nested case statements with helper functions feels like it will create an unnecessarily large footprint.  For most cases, you could just call NumberFormat.getInstance() once, after choosing an appropriate value for the ICU style.  Then you could have some special cases before and after for the weird cases.

Reply:
Fixed most of the clutter in the latest iteration .. Please take a look and let me know.